svm/src/account_loader.rs

```diff
@@ -283,95 +287,50 @@ fn load_transaction_accounts<CB: TransactionProcessingCallback>(
         let mut account_found = true;
-        let is_instruction_account = u8::try_from(i)
-            .map(|i| instruction_accounts.contains(&&i))
-            .unwrap_or(false);
-        let (account_size, account, rent) = if solana_sdk::sysvar::instructions::check_id(key) {
-            // Since the instructions sysvar is constructed by the SVM
-            // and modified for each transaction instruction, it cannot
-            // be overridden.
-            (
-                0, /* loaded size */
-                construct_instructions_account(message),
-                0, /* collected rent */
-            )
-        } else if let Some(account_override) =
-            account_overrides.and_then(|overrides| overrides.get(key))
-        {
-            (account_override.data().len(), account_override.clone(), 0)
-        } else if let Some(program) = (!is_instruction_account && !message.is_writable(i))
-            .then_some(())
-            .and_then(|_| loaded_programs.find(key))
-        {
-            callbacks
-                .get_account_shared_data(key)
-                .ok_or(TransactionError::AccountNotFound)?;
-            // Optimization to skip loading of accounts which are only used as
-            // programs in top-level instructions and not passed as instruction accounts.
-            let program_account = account_shared_data_from_program(&program);
-            (program.account_size, program_account, 0)
-        } else {
-            callbacks
-                .get_account_shared_data(key)
-                .map(|mut account| {
-                    if message.is_writable(i) {
-                        let rent_due = collect_rent_from_account(
-                            feature_set,
-                            rent_collector,
-                            key,
-                            &mut account,
-                        )
-                        .rent_amount;
-
-                        (account.data().len(), account, rent_due)
-                    } else {
-                        (account.data().len(), account, 0)
-                    }
-                })
-                .unwrap_or_else(|| {
-                    account_found = false;
-                    let mut default_account = AccountSharedData::default();
-                    // All new accounts must be rent-exempt (enforced in Bank::execute_loaded_transaction).
-                    // Currently, rent collection sets rent_epoch to u64::MAX, but initializing the account
-                    // with this field already set would allow us to skip rent collection for these accounts.
-                    default_account.set_rent_epoch(RENT_EXEMPT_RENT_EPOCH);
-                    (default_account.data().len(), default_account, 0)
-                })
-        };
-
-        collect_account(key, account_size, account, rent, account_found)?;
+        let loaded_account = load_transaction_account(
```

Comment:
It seems a odd for this to take `account_found` as a mutable instead of returning a tuple `(loaded_account, account_found)`. If I'm missing some context on the choice lmk.

Reply:
Hmm not sure why I decided to that, switching to return a tuple as you suggested.